### PR TITLE
Add support for navigating the taxonomy path through input box

### LINKED ISSUE #89: Enhance file path to navigate through the taxonomy tree for skill and knowledge
Currently file path input box in the skill and knowledge form requires user to provide the path in the taxonomy tree, where they want to add their skill or knowledge. Currently this process is pretty manual. User needs to go through the taxonomy and determine the directory in the tree structure where the skills can be added and provide that path in the file path input box. Overall user experience of current implementation is not very user friendly. Having an option where user don't have to leave the form to determine the taxonomy path, would be the desirable.

### PR DESCRIPTION
**Pull request by vishnoianil**
This PR adds support for navigating the taxonomy tree for skill and knowledge contribution through the form's input box itself. User don't have to leave the skill or knowledge form to navigate through the taxonomy github repo to determine the path where they can place the skill or knowledge.

Fixes #89

### CODE REVIEW

File: .github/workflows/images.yml [hunk 1]
```diff
@@ -10,6 +10,8 @@ env:
   GHCR_UI_IMAGE_NAME: ${{ github.repository }}/ui
   QUAY_REGISTRY: quay.io
   QUAY_UI_IMAGE_NAME: instructlab-ui/ui
+  GHCR_PATHSERVICE_IMAGE_NAME: ${{ github.repository }}/pathservice
```
Comment (aevo98765) [inline .github/workflows/images.yml:13]: If I understand the functionality properly. This service provides people with the directory structure of the taxonomy so they can identify where there new contribution sits. I am not sure `pathservice` is the best description of this? taxonomy-descripion-service?

**Comment by aevo98765** [comment]
I think there would be a strong argument for this service sitting outside of the UI. Could/should this be it's own repo within the InstructLab organisation?

File: .github/workflows/images.yml (hunk 1 above)
Comment (vishnoianil) [reply .github/workflows/images.yml:13]: Yes, it's basically provides you a list of first level directories within given directory in taxonomy repo. Which will be used by the pathservice ui component to provide the navigation feature. Naming is hard. This service helps user to find the path where they can place their files, hence the pathservice. Not sure if taxonomy-description-service fits the intent as well. But i am open to any name that makes sense.

**Comment by vishnoianil** [comment]
> I think there would be a strong argument for this service sitting outside of the UI. Could/should this be it's own repo within the InstructLab organisation?

This is a backend component for the UI service, so in my opinion it makes sense for it to be part of UI project. Also I don't see any other project in instructlab that would like to consume this service at this point of time. If we see interest from outside of the UI project, i think it would be good to explore how best we can expose it as a full fledge service with stable APIs.

File: .github/workflows/images.yml (hunk 1 above)
Comment (aevo98765) [reply .github/workflows/images.yml:13]: Yeh see what you mean. Maybe taxonomy-path-service?

**Comment by aevo98765** [comment]
LGTM.

**Review by aevo98765 — APPROVED** [review]
LGTM

**Pull request merged**